Append metadata

prasar-ashutosh opened the pull request:
#### What type of PR is this?

<!--
Choose one of the following labels :
- Improvement
- Bug Fix
-->

#### What does this PR do / why is it needed ?

<!--
Describe change being introduced by this PR.
-->

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #

#### Other notes for reviewers:

#### Does this PR introduce a user-facing change?
<!--
-->

abhishoya-gs committed:
Prepend suite name to test name in Relational_DbSpecific (#2213)

hardikmaheshwari committed:
Delete UrlStreamExecutionNode from pure code (#2234)

kevin-m-knight-gs committed:
Set the class loader properly for finding code repositories for static metadata for PureModel (#2238)

hardikmaheshwari committed:
Align equality execution on Engine to Pure (#2235)

gs-jp1 committed:
Legend SQL Assortment of changes (#2219)

gs-jp1 committed:
Legend SQL - Support Expression arguments (#2242)

mrudula-gs committed:
Fix selfJoin on Views (#2237)

Fix selfJoin on Views

mrudula-gs committed:
Fix XStore with GraphFetchChecked (#2240)

MauricioUyaguari committed:
fix input for target package in db to model (#2241)

AFine-gs committed:
exclude String list from ParameterizedValueSpecification (#2207)

exclude String list from ParameterizedValueSpecification and add tests

AFine-gs committed:
add initial support for externalize on TDS (#2239)

add initial support for externalize on TDS

hardikmaheshwari committed:
Fix test data generation algorithm to correctly generate quotes and include all views (#2203)

hardikmaheshwari committed:
Fix number overflow issues with adjust operations on date (#2220)

* Fix number overflow issues with adjust operations on date

* Address review

* Apply suggestions from code review

Co-authored-by: Kevin Knight <57677197+kevin-m-knight-gs@users.noreply.github.com>

* Update legend-pure version

---------

Co-authored-by: Kevin Knight <57677197+kevin-m-knight-gs@users.noreply.github.com>

siaka-Akash committed:
Fix test data generation for nested views (#2236)

* fix test data generation for nested views

* add test

* Refactor and add alloy test

* used already existing mainTable function

* refactor

* change copyright year

gayathrir11 committed:
Fix test failure in build (#2247)

finos-admin committed:
[maven-release-plugin] prepare release legend-engine-4.27.0

finos-admin committed:
[maven-release-plugin] prepare for next development iteration

gs-jp1 committed:
SQL - add hashing function support (#2248)

prasar-ashutosh committed:
Add Support for metadata for BulkLoad Task

gs-ssh16 committed:
Change composite PK not null check generated in graph fetch to AND (#2245)

abhishoya-gs committed:
GraphQL Introspection - description, enums and Number type (#2244)

sprisha committed:
remove assert from CompositeExecutionPlan (#2255)

hardikmaheshwari committed:
Remove RoutedVariablePlaceHolder processing from store routing (#2256)

sahil37 committed:
add back mastery acquisition spec in backwards compatible way (#2243)

akphi committed:
connection factory: refactor (#2253)

* minor cleanups

* connection: add test for postgres connection factory

* connection: add DEPRECATED__ prefix to older flow usage

* connection: move connection manager code to a separate module

* connection-man: new API

* minor cleanups

* conn-man: refactor connection factory flows

* minor fixes

* conn-man: revert DEPRECATION__ prefix

finos-admin committed:
[maven-release-plugin] prepare release legend-engine-4.27.1

finos-admin committed:
[maven-release-plugin] prepare for next development iteration

gs-jp1 committed:
Legend SQL - handle tableToTDS functions (#2259)

* Legend SQL - handle tableToTDS functions

* Handle missing rawType case

prasar-ashutosh committed:
Refactor Digest Generation Strategy

kumuwu commented on legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BulkLoadPlanner.java:
```diff
@@ -154,4 +162,58 @@ private Selection getRowsBasedOnAppendTimestamp(Dataset dataset, String field, S
+
+    public static String jsonifyBatchSourceInfo(List<String> files)
+    {
+        Map batchSourceMap = new HashMap();
```
`Map<String, List<String>>`?

prasar-ashutosh replied:
Made it a Map<String, Object>

kumuwu commented on legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BulkLoadPlanner.java:
```diff
@@ -154,4 +162,58 @@ private Selection getRowsBasedOnAppendTimestamp(Dataset dataset, String field, S
         return Selection.builder().source(dataset.datasetReference()).condition(condition).addFields(countFunction).build();
     }
+
+    public static String jsonifyBatchSourceInfo(List<String> files)
```
Would LogicalPlanUtils be a better place to hold this?

prasar-ashutosh replied:
Made it private non static

kumuwu commented on legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BulkLoadPlanner.java:
```diff
@@ -154,4 +162,58 @@ private Selection getRowsBasedOnAppendTimestamp(Dataset dataset, String field, S
+    }
+
+
+    static class DigestGeneration implements DigestGenStrategyVisitor<Void>
```
Would LogicalPlanUtils be a better place to hold this?

prasar-ashutosh replied:
It's too specific to BulkLoadPlanner, let's keep it here

kumuwu commented on legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/Planner.java:
```diff
@@ -26,6 +26,7 @@
 import org.finos.legend.engine.persistence.components.logicalplan.LogicalPlanFactory;
 import org.finos.legend.engine.persistence.components.logicalplan.conditions.Condition;
 import org.finos.legend.engine.persistence.components.logicalplan.datasets.Dataset;
+import org.finos.legend.engine.persistence.components.util.AppendLogMetadataDataset;
```
The position of this import not following alphabetical order

prasar-ashutosh replied:
Moved

kumuwu commented on legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-ansi/src/test/java/org/finos/legend/engine/persistence/components/util/AppendLogDatasetUtilsTest.java:
```diff
@@ -0,0 +1,87 @@
+    {
+        AppendLogMetadataUtils appendLogMetadataUtils = new AppendLogMetadataUtils(appendLogMetadataDataset);
+        StringValue batchIdValue = StringValue.of("batch_id_123");
+        StringValue appendLogTableName = StringValue.of("appeng_log_table_name");
```
To be a more realistic test for upper case, this StringValue should be "APPEND_LOG_TABLE_NAME"

rafaelbey committed:
Add decimal and numeric to dyna function type inference (#2254)

sahil37 committed:
add additional mastery spec for processing (#2260)

rafaelbey committed:
Retry metadata calls on some http response codes (#2261)

gs-jp1 committed:
Datediff millisecond in snowflake (#2264)

prasar-ashutosh committed:
Addressed Code Review Comments

hardikmaheshwari committed:
Add checked and defaultDefectTree function handlers (#2263)

gs-ssh16 committed:
Delete deprecated local platform binding aspects (#2262)

gs-jp1 committed:
ElasticSearch - IndexToTDS schema resolve (#2265)

prasar-ashutosh committed:
Refactor Code to rename AppendLog to BulkLoad

hardikmaheshwari committed:
Pre-compute declared methods to optimization deserialization of json (#2268)

prasar-ashutosh committed:
Add default bulkLoad Batchid

prasar-ashutosh committed:
Refactor Append Log table name

hardikmaheshwari committed:
Fix reference to allocationNode created for views (#2267)

Commit:
M2m2r milestoning support (#2176)

* add m2m2r milestoning support
- change scan properties for milestoned qualified property
- add support for milestoned source to non-milestoned target in m2m via no-arg property
- propagate parent Milestoning context for no arg properties in relational graph fetch
- minor refactor

* Create milestonedSourceToMilestonedTargetProperty.pure

* Create milestonedSourceToNonMilestonedTargetProperty.pure

* Create nonMilestonedSourceToMilestonedTargetProperty.pure

* Create shared.pure

* Delete legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/modelToModelToRelational/milestoned_graphFetch_m2m2r.pure

* use flag explodeMilestonedProperties

* Update graphExtension.pure

hardikmaheshwari committed:
Fix test data generation for queries involving views (#2272)

gs-manvig committed:
Fix `is not distinct from` sql-to-pure (#2269)

* Remove redundant `isTrue`

sahil37 committed:
adding mastery decryption protocol extension (#2275)

Yasirmod17 committed:
Uplift Service Ownership (#2274)

kumuwu committed:
Persistence Component: Bulk Load Support for H2 (#2210)

* Bug Fix: Bitemporal milestoning Derive Main schema removes the VALID_FROM/VALID_TRHOUGH field if the name matches with source specified VALID_FROM/VALID_TRHOUGH fields

* Bug Fix: Bitemporal milestoning Schema Evolution must ignore user provided validity fields

* Adding code for concurrent safety feature

* Adding test for Multi Ingest Mode with concurrent Safety

* Adding tests for concurrent safety

* Code Clean up

* Bug Fix: Bitemporal temp tables must be deleted after usage

* Update readme and code review comments

* Fix typo

* Fix typos in readme

* Bug Fix: Empty Batch Handling in Unitemp Snapshot

* Bug Fix: Code review comments

* Implement StagedFilesDatasetReference

* Implement StagedFilesSelection

* Support for Empty Batch Handling in Unitemporal Snapshot

* Support for FailEmptyBatch strategy in Unitemporal Snapshot

* Enrich datasets to add additionalDatasetproperties every where

* Add tests for Empty Data handling

* Support ICEBERG_TABLE_2022 for Iceberg tables

* Implement bulk load for h2

* Add missing datasetAdditionalProperties in BigQueryHelper

* Implement executor flow and add end-to-end test

* Add test

* Add support for digest udf in h2 and add tests

* Clean up

* Add file format and validation for file format and add tests

* Add missing statement

* Fix typo in year

* Fix comments

* Add H2 MD5

* Change stats

---------

Co-authored-by: Ashutosh <prasar.ashutosh@gmail.com>
Co-authored-by: prasar-ashutosh <ashutosh.prasar@hk.email.gs.com>

rafaelbey committed:
Update legend-pure to 4.6.0 (#2276)

Update legend-pure to 4.6.0

AFine-gs committed:
skip freemarker when there are no templates (#2250)

skip freemarker when there are no templates

finos-admin committed:
[maven-release-plugin] prepare release legend-engine-4.28.0

finos-admin committed:
[maven-release-plugin] prepare for next development iteration

irisyngao committed:
Test Data Generation - only show necessary columns of tables used in query (#2202)

* Test Data Gen - just show necessary columns of tables used in query

* resolve comments

gs-jp1 committed:
Legend SQL - assortment of updates (#2278)

- handle nullable args in function translations
- small int cast support
- fix some generic types

prasar-ashutosh committed:
Merge remote-tracking branch 'origin/master' into append_metadata

# Conflicts:
#	legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BulkLoadPlanner.java
#	legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/java/org/finos/legend/engine/persistence/components/ingestmode/bulkload/BulkLoadTest.java
#	legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/resources/data/bulk-load/expected/expected_table1.csv
#	legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/resources/data/bulk-load/expected/expected_table2.csv
#	legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/resources/data/bulk-load/expected/expected_table3.csv
#	legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-h2/src/test/resources/data/bulk-load/expected/expected_table4.csv
